os/drivers/wireless/realtek/wifi_util_interface/wifi_interactive_mode.c

```diff
@@ -772,91 +772,107 @@ static void print_scan_result(rtw_scan_result_t *record)
+		scan_list->ap_info.max_rate = 0;
+		scan_list->ap_info.rssi = record->signal_strength;
+		scan_list->ap_info.phy_mode = 0x00000004; //bit2 is set to 1 if support 11n
 		switch (record->security) {
 		case RTW_SECURITY_OPEN:
-			_scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_OPEN;
+			scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_OPEN;
 			break;
 		case RTW_SECURITY_WEP_SHARED:
-			_scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_WEP_SHARED;
+			scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_WEP_SHARED;
 			break;
 		case RTW_SECURITY_WEP_PSK:
-			_scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_WPA_PSK;
+			scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_WPA_PSK;
 			break;
 		case RTW_SECURITY_WPA2_AES_PSK:
-			_scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_WPA2_PSK;
+			scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_WPA2_PSK;
 			break;
 		case RTW_SECURITY_WPA_WPA2_MIXED:
-			_scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_WPA_AND_WPA2_PSK;
+			scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_WPA_AND_WPA2_PSK;
 			break;
-		//case WIFI_AUTH_WPA2_ENTERPRISE:
 		default:
-			_scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_UNKNOWN;
+			scan_list->ap_info.ap_auth_type = WIFI_UTILS_AUTH_UNKNOWN;
 			break;
 		}
 
 		switch (record->security) {
 		case RTW_SECURITY_OPEN:
-			_scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_NONE;
+			scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_NONE;
 			break;
 		case RTW_SECURITY_WEP_SHARED:
-			_scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_WEP_64;
+			scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_WEP_64;
 			break;
 		case RTW_SECURITY_WEP_PSK:
-			_scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_WEP_128;
+			scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_WEP_128;
 			break;
 		case RTW_SECURITY_WPA_TKIP_PSK:
-			_scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP;
+			scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP;
 			break;
 		case RTW_SECURITY_WPA_AES_PSK:
 		case RTW_SECURITY_WPA2_AES_PSK:
-			_scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_AES;
+			scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_AES;
 			break;
 		case RTW_SECURITY_WPA2_MIXED_PSK:
-			_scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP_AND_AES;
+			scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP_AND_AES;
 			break;
 		default:
-			_scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_UNKNOWN;
+			scan_list->ap_info.ap_crypto_type = WIFI_UTILS_CRYPTO_UNKNOWN;
 			break;
 		}
-
-		if (i > 0) {
-			(_scan_list - 1)->next = _scan_list;
-			_scan_list->next = NULL;
+		if (g_scan_list == NULL) {
+			g_scan_list = scan_list;
+			g_scan_num = 1;
+		} else {
+			wifi_utils_scan_list_s *cur = g_scan_list;
+			int idx;
```

Comment:
declaring variable just before you need it is correct ?
OR, all variable should be declared as the function starts.